src/test/java/cloud/prefab/client/IntegrationTest.java

```diff
@@ -0,0 +1,81 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import cloud.prefab.client.integration.DynamicTestBuilder;
+import cloud.prefab.client.integration.IntegrationTestFile;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.function.Executable;
+import org.yaml.snakeyaml.Yaml;
+
+class IntegrationTest {
+
+  static String INTEGRATION_TEST_DATA_DIRECTORY =
+    "./src/test/resources/prefab-cloud-integration-test-data";
+
+  String getIntegrationTestsVersion() throws IOException {
+    Path path = Paths.get(INTEGRATION_TEST_DATA_DIRECTORY + "/version");
+
+    String version = Files.readAllLines(path).get(0);
+    return version;
+  }
+
+  File[] findIntegrationTestFiles() {
+    try {
+      File dir = new File(
+        INTEGRATION_TEST_DATA_DIRECTORY + "/tests/" + getIntegrationTestsVersion()
+      );
+      File[] files = dir.listFiles();
+      return files;
+    } catch (IOException e) {
+      return new File[0];
```

Comment:
I would `throw new RuntimeException(e)`  I don't think you really want to carry on.